lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1689,7 +1689,15 @@ def __init__(
         self._relation_name = relation_name
         self._container = self._get_container(container_name)
         self._container_name = self._get_container_name(container_name)
-        self._log_files = log_files or []
+
+        if log_files is None:
+            log_files = []
```

Comment:
The only difference here is that with the new way, if the user passes something like `""` or `{}` we now raise and exception rather than silently converting to a list. Which behavior is preferred? I'm not sure.

Reply:
Correct, anything false-y would have become `[]`.
- Seems odd though that for `log_files=""` it works but then `log_files="file"` is a TypeError.
- We could accept `Union[list, set, tuple]` but just one is simpler and list seems most appropriate.

Reply:
Silently converting to a list is preferable so we don't raise brand-new exceptions on a `fetch-lib` with code that used to work.

```python
if log_files and isinstance(log_files, str):
  log_files = [log_files]
elif not isinstance(log_files, (list, set, tuple)) and log_files is None:
  log_files = []
else:
  logger.error("Some message about how it's invalid, and that we're going to zero it out, but it won't work")
  log_files = []
```

We can still warn the author in a manner far less disruptive than raising exceptions for previously acceptable behavior and making them run it down (and potentially rewriting their unit tests before they can continue).

It's not as cut-and-dry, but we live with it until v1.

Reply:
Done